From 878f9fe360a59796f53868bcb311a435ac8e6ee4 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 24 May 2018 13:44:30 +0300
Subject: [PATCH 0628/1051] dts: mvebu: Update A8K AP806 SDHCI settings

Update the settings for AP806 SDHCI interface according to
latest Xenon drivers changes.
- no need to select the PHY slow mode anymore
- recommended to add HS400 support at 1.8V signalling on AP806-B0
- fix the bus witdth for A8040 DB from 4 to 8 bits.

Change-Id: Ibdd07dbdfababb9158f3fe0996c382ff3009ff84
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db.dts    | 10 +++++++++-
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts |  5 -----
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi     |  1 -
 3 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dts b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
index 7b766125682a..ee89d0e883e9 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
@@ -360,7 +360,15 @@
 
 &ap_sdhci0 {
 	status = "okay";
-	bus-width = <4>;
+	bus-width = <8>;
+	/* The below property should be added to boards with AP806-B0
+	 * for enabling HS400 speed mode. Otherwise the device highest
+	 * speed mode will be HS200.
+	 * Should not be added to boards with earlier release of AP806
+	 * since it will cause SDHCI driver to fail upon initialization.
+	 *
+	 * mmc-hs400-1_8v;
+	 */
 	non-removable;
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index e7aceed8c054..9319d0be507a 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -111,11 +111,6 @@
 
 &ap_sdhci0 {
 	bus-width = <8>;
-	/*
-	 * Not stable in HS modes - phy needs "more calibration", so add
-	 * the "slow-mode" and disable SDR104, SDR50 and DDR50 modes.
-	 */
-	marvell,xenon-phy-slow-mode;
 	no-1-8-v;
 	no-sd;
 	no-sdio;
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 985eb59dfd22..7b05bee74ff8 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -235,7 +235,6 @@
 				clock-names = "core";
 				clocks = <&ap_clk 4>;
 				dma-coherent;
-				marvell,xenon-phy-slow-mode;
 				status = "disabled";
 			};
 
-- 
2.17.1

